### ISSUE TITLE
Best approach to validation

### ISSUE THREAD

**Issue by reinink** [thephpleague/glide#2]
Currently Glide does all validation before any image manipulations are performed. This allows for quick validation of the request without having to actually perform any manipulations. It also allows for [nice error pages](http://glide.herokuapp.com/kayaks.jpg?w=a&filt=wrong), that show multiple errors.

However, this has one major problem. Each individual [manipulator](https://github.com/reinink/glide/tree/master/src/Manipulators) is isolated and runs their validation blind to what other manipulators may have done before it. For example, the current [Size](https://github.com/reinink/glide/blob/master/src/Manipulators/Size.php) manipulator validates an image crop against the source image size. However, it's possible that a previous manipulator changed the size already, but since no actual manipulations were performed in the validation stage, the new size is not known.

I suggest moving to an ignore/exception based error validation system. Any invalid function requests will be simply ignored, unless they absolutely cannot be, in which case an exception is thrown. This is actually how Imgix and Cloudinary both work.

Further, the current "soft" validation has issues when it comes to extending Glide. For example, the [Effects](https://github.com/reinink/glide/blob/master/src/Manipulators/Effects.php) manipulator triggers an error if the filter (`filt`) function isn't `greyscale` or `sepia`. However, an ignore based approach would allow a subsequent manipulator to fulfill that request.

In summary, I propose:
- Remove the "soft" validation stage, validation must performed against an actual image.
- Whenever possible, simply ignore functions that have invalid parameters.
- If it's an error that cannot be ignored, throw an exception and stop execution.
